src/main/java/net/dv8tion/jda/api/utils/messages/MessageCreateBuilder.java

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright 2015 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.api.utils.messages;
+
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.interactions.components.LayoutComponent;
+import net.dv8tion.jda.api.utils.FileUpload;
+import net.dv8tion.jda.internal.utils.AllowedMentionsImpl;
+import net.dv8tion.jda.internal.utils.Checks;
+import net.dv8tion.jda.internal.utils.Helpers;
+import net.dv8tion.jda.internal.utils.IOUtil;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Builder specialized for building a {@link MessageCreateData}.
+ *
+ * @see MessageEditBuilder
+ */
```

Comment:
As a note, the MessageBuilder used to live at the root of JDA at `net.dv8tion.jda.api.MessageBuilder` but these new builders live deep inside of utils. Not sure what to think about that yet.

Reply:
Actually, it looks like not just Builders, but the Data and Request interfaces live there too. Makes me think we need to move this out of `utils` and into a closer to top-level package. We have a looot of message related stuff
 - `MessageType`
 - `MessageHistory`
 - `MessageReference`
 - `MessageEmbed`
 - `MessageReaction`
 - `Message`

Maybe it would be better to introduce
 - `net.dv8tion.jda.api.entities.message`
 - `net.dv8tion.jda.api.entities.message.data`
 - `net.dv8tion.jda.api.entities.message.builder`
 
 Or something similar?

Reply:
Builders make more sense here in this util package, they aren't really entities or related to those listed types beside the name.